### PR TITLE
Resolving issue #375 StorageClass header is not added to upload request

### PR DESCRIPTION
**Pull request by padamuscinova**
Added possibility to set StorageClass in ObjectMetadata (setStorageClass(StorageClass storageClass))
Save value into DB (TransferRecord). Table adjusted.
Adjusted tests to test also set StorageClass.

### CODE REVIEW

**Comment by kvasukib** [comment]
Hi @Petra, Thank you for submitting a pull request. We are reviewing it.

**Comment by kvasukib** [comment]
Sorry for the delayed response. Can you remove the unused import of `StorageClass` from `TransferRecord.java`? There seem to be some extra whitespaces in some of the code changes. Can you remove them?

**Comment by padamuscinova** [comment]
@kvasukib I've removed the unused import and reformatted all touched classes with your "AWS Mobile SDK for Android" formatter. Seems like the code was not formatted properly before my changes thus some additional formatting was changed not only whitespaces.

**Comment by padamuscinova** [comment]
Does anybody care to merge this after 4 months?

**Comment by kvasukib** [comment]
Hi @padamuscinova, Sorry for the delay. I am working on it.

**Comment by kvasukib** [comment]
I was able to reproduce the issue. The issue is that the S3 putObject method extracts the storage class from the put object request and adds it as an Http header to the underlying request object. However, the TransferUtility does not have a provision to accept the storage class directly and we rely on the customer user metadata which is added to the `ObjectMetadata`. The UploadTask in TransferUtility does not add the storage class to the put object request.

I made a change to add the storage class to the put object request when it's been part of the custom user-metadata and found it to be working for the 4 storage classes except for Glacier.

@padamuscinova Were you able to set Glacier storage class and test it?

File: aws-android-sdk-s3/src/test/java/com/amazonaws/services/s3/model/ObjectMetadataTest.java [hunk 1]
```diff
@@ -77,6 +79,7 @@ public void cloneNonEmpty() {
         from.setHeader(Headers.CONTENT_RANGE, "/9999");
         from.setHeader(Headers.S3_VERSION_ID, "versionid");
         from.setHeader(Headers.ETAG, "etag");
+        from.setStorageClass(StorageClass.ReducedRedundancy);
```
Comment (kvasukib) [inline aws-android-sdk-s3/src/test/java/com/amazonaws/services/s3/model/ObjectMetadataTest.java]: Unfortunately, there is no method called setStorageClass. Instead, can you use setHeader with Headers.STORAGE_CLASS?

```
from.setHeader(Headers.STORAGE_CLASS, StorageClass.ReducedRedundancy.toString());
```

**Comment by kvasukib** [comment]
Approving the PR. Will merge it in the upcoming release.

**Pull request merged**